Lib/test/test_type_params.py

```diff
@@ -0,0 +1,558 @@
+            type_params[0].__bound__
+        self.assertEqual(type_params[0].__constraints__, None)
+        self.assertEqual(type_params[1].__bound__, None)
+        with self.assertRaises(NameError):
+            type_params[1].__constraints__
+
+        Undefined = "defined"
+        self.assertEqual(type_params[0].__bound__, "defined")
+        self.assertEqual(type_params[0].__constraints__, None)
+
+        self.assertEqual(type_params[1].__bound__, None)
+        self.assertEqual(type_params[1].__constraints__, ("defined",))
+
+
+class TypeParamsClassScopeTest(unittest.TestCase):
+    def test_alias(self):
+        class X:
+            T = int
+            type U = T
+        self.assertIs(X.U.__value__, int)
+
+        code = textwrap.dedent("""\
+            glb = "global"
+            class X:
+                cls = "class"
+                type U = (glb, cls)
+
+            assert X.U.__value__ == ("global", "class"), X.U.__value__
+        """)
+        exec(code, {})
+
+    def test_bound(self):
+        class X:
+            T = int
+            def foo[U: T](self): ...
+        self.assertIs(X.foo.__type_params__[0].__bound__, int)
+
+        code = textwrap.dedent("""\
+            glb = "global"
+            class X:
+                cls = "class"
+                def foo[T: glb, U: cls](self): ...
+
+            T, U = X.foo.__type_params__
+            assert T.__bound__ == "global"
+            assert U.__bound__ == "class"
+        """)
+        exec(code, {})
+
+
+class ManglingTest(unittest.TestCase):
+    def test_mangling(self):
+        class Foo[__T]:
+            param = __T
+            def meth[__U](self, arg: __T, arg2: __U):
+                return (__T, __U)
+            type Alias[__V] = (__T, __V)
+
+        T = Foo.__type_params__[0]
+        self.assertEqual(T.__name__, "__T")
+        U = Foo.meth.__type_params__[0]
+        self.assertEqual(U.__name__, "__U")
+        V = Foo.Alias.__type_params__[0]
+        self.assertEqual(V.__name__, "__V")
+
+        anno = Foo.meth.__annotations__
+        self.assertIs(anno["arg"], T)
+        self.assertIs(anno["arg2"], U)
+        self.assertEqual(Foo().meth(1, 2), (T, U))
+
+        self.assertEqual(Foo.Alias.__value__, (T, V))
+
+
+class TypeParamsTraditionalTypeVars(unittest.TestCase):
+    def test_traditional_01(self):
+        code = textwrap.dedent("""\
+                from typing import Generic
+                class ClassA[T](Generic[T]): ...
+            """
+        )
+
+        with self.assertRaisesRegex(TypeError, r"Cannot inherit from Generic\[...\] multiple types."):
+            exec(code, {})
+
+    def test_traditional_02(self):
+        from typing import TypeVar
+        S = TypeVar("S")
+        with self.assertRaises(TypeError):
+            class ClassA[T](dict[T, S]): ...
+
+    def test_traditional_03(self):
+        # This does not generate a runtime error, but it should be
+        # flagged as an error by type checkers.
```

Comment:
wondering why this is wrong

Reply:
You can't mix pre-PEP 695 and PEP 695 typevars. In the example, only `T` is defined using PEP 695 syntax, but the function is generic over both T and S.

(Also, thanks for the comments! I'll add additional test cases.)

Reply:
> You can't mix pre-PEP 695 and PEP 695 typevars.

Is this a technical issue behind or a design choice? Mix using them looks natural.